server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -3732,11 +3732,9 @@ ReplicationTracker getReplicationTracker() {
     }
 
     /**
-     * Executes a scheduled refresh if necessary.
-     *
-     * @return <code>true</code> iff the engine got refreshed otherwise <code>false</code>
+     * Executes a scheduled refresh if necessary. Completes the listener with true if a refreshed was performed otherwise false.
      */
-    public boolean scheduledRefresh() {
+    public void scheduledRefresh(ActionListener<Boolean> listener) {
```

Comment:
Is there a reason why to have an `ActionListener<Boolean>` instead of `ActionListener<RefreshResult>`?

Reply:
I agree that it would be nice to keep it consistent with refresh and use `ActionListener<RefreshResult>`, and instead of `false`, use `Engine.RefreshResult.NO_REFRESH` to resolve the listener.